crypto/ec/curve448/arch_neon/f_impl.h

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2017-2018 The OpenSSL Project Authors. All Rights Reserved.
+ * Copyright 2014-2016 Cryptography Research, Inc.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ *
+ * Originally written by Mike Hamburg
+ */
+
+#define GF_HEADROOM 2
+#define LIMBPERM(x) (((x)<<1 | (x)>>3) & 15)
+#define USE_NEON_PERM 1
+#define LIMBHI(x) ((x##ull)>>28)
+#define LIMBLO(x) ((x##ull)&((1ull<<28)-1))
+#define FIELD_LITERAL(a,b,c,d,e,f,g,h) { \
+    { \
+        LIMBLO(a), LIMBLO(e), LIMBHI(a), LIMBHI(e), LIMBLO(b), LIMBLO(f), \
+        LIMBHI(b), LIMBHI(f), LIMBLO(c), LIMBLO(g), LIMBHI(c), LIMBHI(g), \
+        LIMBLO(d), LIMBLO(h), LIMBHI(d), LIMBHI(h) \
+    } \
+}
+
+#define LIMB_PLACE_VALUE(i) 28
+
+void gf_add_RAW(gf out, const gf a, const gf b)
+{
+    for (unsigned int i = 0; i < sizeof(*out) / sizeof(uint32xn_t); i++) {
+        ((uint32xn_t *) out)[i] =
```

Comment:
this looks like a strict-aliasing violation

Reply:
The comments in the architecture specific files except anything inside arch_32 have barely been touched by my at all. They are not used or compiled. I only wanted to import them because I thought we would probably integrate them at a later stage. If that is considered a blocker for review I could remove these files completely (but then we would probably add them back in again at some later time).

Reply:
yes, that might be better...

Reply:
I removed this file. It still gets imported in the original commit, and then gets deleted again. This means we can just resurrect the code easily without having to worry about the code's provenance/CLAs etc.